lib/puppet/resource/catalog.rb

```diff
@@ -591,12 +591,12 @@ def fail_on_duplicate_type_and_title(resource, title_key)
     return unless existing_resource = @resource_table[title_key]
 
     # If we've gotten this far, it's a real conflict
-    msg = "Duplicate declaration: #{resource.ref} is already declared"
-
-    msg << " in file #{existing_resource.file}:#{existing_resource.line}" if existing_resource.file and existing_resource.line
-
-    msg << "; cannot redeclare"
-
+    error_location_str = Puppet::Util::Errors.error_location(existing_resource.file, existing_resource.line)
+    msg = if error_location_str.empty?
+            _("Duplicate declaration: %{resource} is already declared; cannot redeclare") % { resource: resource.ref }
+          else
+            _("Duplicate declaration: %{resource} is already declared; at %{error_location} cannot redeclare") % { resource: resource.ref, error_location: error_location_str }
```

Comment:
Looks like the `;` here was originally always before "cannot redeclare" (and after "at &lt;location&gt;" if it was there). Typo in moving it before the location?

Reply:
Arg you are correct. Fixed